ekncontent/ekncontent/eknc-domain.c

```diff
@@ -850,6 +850,56 @@ eknc_domain_query_finish (EkncDomain *self,
 }
 
 /**
+ * eknc_domain_read_uri:
+ * @self: domain
+ * @uri: the ekn uri to read
+ * @mime_type: (out) (transfer none) (allow-none): return location for the content mime type
+ * @error: #GError for error reporting
+ *
+ * Reads the contents of a ekn uri within the domain as a GBytes. Optionally
+ * returns the mime type of the contents as well.
+ *
+ * Returns: (transfer full): the gbytes
+ */
+GBytes *
+eknc_domain_read_uri (EkncDomain *self,
+                      const gchar *uri,
+                      const gchar **mime_type,
+                      GError **error)
+{
+  if (!uri || !g_str_has_prefix (uri, "ekn:"))
+    return NULL;
+
+  g_auto(GStrv) tokens = g_strsplit (uri + 6, "/", -1);
```

Comment:
For clarity use `strlen("ekn://")`, pretty sure any compiler will optimize it back to 6